vector-im/element-android
# [Device Management] Save matrix_client_information events on login/registration

## Type of change


## Content


Adding a new step after the `Session` has been configured and started: update if needed of Matrix client info in `AccountData` which will be used in `Session` details screen to render `Application` section.
A new use case has been introduced `ConfigureAndStartSessionUseCase` which replaces the extension method `Session.configureAndStart()`.

## Motivation and context


Closes #7257 

## Screenshots / GIFs





## Tests



- Login to an account
- Check (using Flipper for example) that the request to update the `AccountData` is correctly sent when client info has changed (app name or version)

## Tested devices

- OS version(s): Android 11

## Checklist
